include/onnxruntime/core/framework/float16.h

```diff
@@ -109,6 +113,19 @@ struct BFloat16 {
 #endif
 };
 
+// User defined suffixes to make it easier to declare
+// initializers with MLFloat16 and BFloat16 from unsigned short
+// E.g 10_f16 or 10_b16
+#if !defined(__CUDACC__) && !defined(__HIPCC__)
+inline MLFloat16 operator"" _f16(unsigned long long int v) {
+  return MLFloat16(gsl::narrow<uint16_t>(v));
```

Comment:
Who is the user for this? Seems only useful if it's called rarely as it's not a cheap operation.

Reply:
See later in the PR